packages/strapi-admin/admin/src/containers/Admin/index.js

```diff
@@ -0,0 +1,386 @@
+    /* istanbul ignore next */
+    ReactGA.initialize(
+      'UA-54313258-9',
+      { testMode: process.env.NODE_ENV === 'test' },
+    );
+
+    // Retrieve the main settings of the application
+    this.props.getInitData();
+  }
+
+  componentDidUpdate(prevProps) {
+    const {
+      admin: { didGetSecuredData, isLoading, isSecured },
+      location: { pathname },
+    } = this.props;
+
+    if (!isLoading && this.state.shouldSecureAfterAllPluginsAreMounted) {
+      if (!this.hasApluginNotReady(this.props)) {
+        this.props.getHook('willSecure');
+      }
+    }
+
+    if (prevProps.location.pathname !== pathname) {
+      this.props.getHook('willSecure');
+
+      /* istanbul ignore if */
+      if (this.isAcceptingTracking()) {
+        ReactGA.pageview(
+          pathname,
+          { testMode: process.env.NODE_ENV === 'test' },
+        );
+      }
+    }
+
+    if (prevProps.admin.isSecured !== isSecured && isSecured) {
+      this.props.getSecuredData();
+    }
+
+    if (prevProps.admin.didGetSecuredData !== didGetSecuredData) {
+      this.props.getHook('didGetSecuredData');
+    }
+  }
+
+  /* istanbul ignore next */
+  componentDidCatch(error, info) {
+    /* eslint-disable */
+    console.log('An error has occured');
+    console.log('--------------------');
+    console.log(error);
+    console.log('Here is some infos');
+    console.log(info);
+    /* eslint-enable */
+
+    // Display the error log component which is not designed yet
+    this.props.setAppError();
+  }
+
+  getContentWrapperStyle = () => {
+    const { admin: { showMenu } } = this.props;
+
+    return showMenu
+      ? { main: {}, sub: styles.content }
+      : { main: { width: '100%' }, sub: styles.wrapper };
+  }
+
+  hasApluginNotReady = props => {
+    const { global: { plugins } } = props;
+
+    return !Object.keys(plugins).every(plugin => (plugins[plugin].isReady === true));
+  } 
+
+  helpers = {
+    hideLeftMenu: this.props.hideLeftMenu,
+    hideLogout: this.props.hideLogout,
+    setAppSecured: this.props.setAppSecured,
+    showLeftMenu: this.props.showLeftMenu,
+    showLogout: this.props.showLogout,
+    unsetAppSecured: this.props.unsetAppSecured,
+    updatePlugin: this.props.updatePlugin,
+  };
+
+  isAcceptingTracking = () => {
+    const { admin: { uuid } } = this.props;
+
+    return !!uuid;
+  }
+
+  /**
+   * Display the app loader until the app is ready
+   * @returns {Boolean}
+   */
+  showLoader = () => {
+    const {
+      admin: { isLoading },
+      global: { isAppLoading },
+    } = this.props;
+
+    if (isAppLoading) {
+      return true;
+    }
+
+    if (isLoading) {
+      return true;
+    }
+
+    return this.hasApluginNotReady(this.props);
+  }
+
+  renderInitializers = () => {
+    const {
+      global: { plugins },
+    } = this.props;
+
+    return Object.keys(plugins).reduce((acc, current) => {
+      const Compo = plugins[current].initializer;
```

Comment:
We could have a better naming like `InitializerComponent`.

Reply:
https://github.com/strapi/strapi/blob/f5ab26fb466741abc9d1d4e60c2788840001bd12/packages/strapi-admin/admin/src/containers/Admin/index.js#L199-L208